test/testsuppressions.cpp

```diff
@@ -425,17 +425,27 @@ class TestSuppressions : public TestFixture {
     }
 
     void inlinesuppress() {
-        Suppressions::Suppression s;
+        std::vector<Suppressions::Suppression> s;
         std::string msg;
-        ASSERT_EQUALS(false, s.parseComment("/* some text */", &msg));
-        ASSERT_EQUALS(false, s.parseComment("/* cppcheck-suppress */", &msg));
+        ASSERT_EQUALS(false, Suppressions::Suppression::parseComment("/* some text */", &s, &msg));
+        ASSERT_EQUALS(0, s.size());
+        ASSERT_EQUALS("", msg);
 
+        ASSERT_EQUALS(false, Suppressions::Suppression::parseComment("/* cppcheck-suppress */", &s, &msg));
+        ASSERT_EQUALS(0, s.size());
+        ASSERT_EQUALS("Could not find attribute for suppression in comment: '/* cppcheck-suppress */'. "
+                      "Make sure to follow the following pattern: "
+                      "'// cppcheck-suppress <attribute> [ symbolName=<restriction> ] [ <attribute> [ symbolName=<restriction> ] ... ] [; comment text]'", msg);
         msg.clear();
-        ASSERT_EQUALS(true, s.parseComment("/* cppcheck-suppress id */", &msg));
+
+        ASSERT_EQUALS(true, Suppressions::Suppression::parseComment("/* cppcheck-suppress id */", &s, &msg));
+        ASSERT_EQUALS(1, s.size());
         ASSERT_EQUALS("", msg);
 
-        ASSERT_EQUALS(true, s.parseComment("/* cppcheck-suppress id some text */", &msg));
-        ASSERT_EQUALS("Bad suppression attribute 'some'. You can write comments in the comment after a ; or //. Valid suppression attributes; symbolName=sym", msg);
+        s.clear();
+        ASSERT_EQUALS(true, Suppressions::Suppression::parseComment("/* cppcheck-suppress id some text */", &s, &msg));
```

Comment:
@danmar I think this syntax with suppression ids in parentheses is a good choice, because it more readable and allows user to annotate reason of this suppression. For example:

```c
// cppcheck-suppress [ unusedFunction ] not yet implemented in this release
```

Similar approach is used in [clang-tidy](https://clang.llvm.org/extra/clang-tidy/#suppressing-undesired-diagnostics).

Reply:
Extending the syntax for example for multi-id suppression with parentheses sounds good.
But IMO full backward compatibility must be guaranteed.
We could enable inline suppressions for daca@home to hopefully find possible issues, maybe there are projects using it.

Reply:
Yes we must be backward compatible. Existing suppressions must work as before.